File: app/src/main/java/com/example/edulockapp/ui/splashscreen/SplashScreen.java
package com.example.edulockapp.ui.splashscreen;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;

import androidx.appcompat.app.AppCompatActivity;

import com.example.edulockapp.R;
import com.example.edulockapp.ui.onboardingscreen.IntroActivity;
import com.example.edulockapp.utils.Utils;

public class SplashScreen extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.splash_screen);

        String EXTRA_LAST_APP = "EXTRA_LAST_APP";
        new Utils(this).setLastApp(EXTRA_LAST_APP);

        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {

                Intent introStart = new Intent(getApplicationContext(), IntroActivity.class);
                startActivity(introStart);
                finish();
            }
        }, 2000);
    }
}

File: app/src/main/java/com/example/edulock/onboarding/splashScreen.kt
package com.example.edulock.onboarding

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import com.example.edulock.R

class splashScreen : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash_screen)

        Handler(Looper.getMainLooper()).postDelayed({
            // Delay and Start Activity
            val intent = Intent(this,MainActivity::class.java)
            startActivity(intent)
            finish()
        } , 2000)
    }

}